pkg/sources_server_client/client.go

```diff
@@ -0,0 +1,101 @@
+package sources_server_client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+)
+
+type VersionPayload struct {
+	app      *v1alpha1.Application
+	revision *string
+}
+
+type DependenciesMap struct {
+	Lock         string `json:"helm/Chart.lock"`
+	Deps         string `json:"helm/dependencies"`
+	Requirements string `json:"helm/requirements.yaml"`
+}
+
+type AppVersionResult struct {
+	AppVersion   string          `json:"appVersion"`
+	Dependencies DependenciesMap `json:"dependencies"`
+}
+
+type SourcesServerConfig struct {
+	BaseURL string
+}
+
+type sourceServerClient struct {
+	clientConfig *SourcesServerConfig
+}
+
+type SourceServerClientInteface interface {
+	GetAppVersion(app *v1alpha1.Application, revisions *string) *AppVersionResult
+}
+
+func (c *sourceServerClient) sendRequest(method, url string, payload interface{}) ([]byte, error) {
```

Comment:
method type missing

Reply:
It is a `string` for both `method` and `url`